warehouse/integrations/postgres/load.go

```diff
@@ -18,172 +21,231 @@ import (
 	"github.com/lib/pq"
 	"golang.org/x/exp/slices"
 
-	"github.com/rudderlabs/rudder-go-kit/stats"
 	"github.com/rudderlabs/rudder-server/warehouse/logfield"
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
-type loadTableResponse struct {
-	StagingTableName string
-}
-
 type loadUsersTableResponse struct {
 	identifiesError error
 	usersError      error
 }
 
-func (pg *Postgres) LoadTable(ctx context.Context, tableName string) error {
-	pg.logger.Infow("started loading",
+func (pg *Postgres) LoadTable(ctx context.Context, tableName string) (*types.LoadTableStats, error) {
+	log := pg.logger.With(
 		logfield.SourceID, pg.Warehouse.Source.ID,
 		logfield.SourceType, pg.Warehouse.Source.SourceDefinition.Name,
 		logfield.DestinationID, pg.Warehouse.Destination.ID,
 		logfield.DestinationType, pg.Warehouse.Destination.DestinationDefinition.Name,
 		logfield.WorkspaceID, pg.Warehouse.WorkspaceID,
 		logfield.Namespace, pg.Namespace,
 		logfield.TableName, tableName,
+		logfield.LoadTableStrategy, pg.loadTableStrategy(),
 	)
+	log.Infow("started loading")
 
-	err := pg.DB.WithTx(ctx, func(tx *sqlmiddleware.Tx) error {
-		tableSchemaInUpload := pg.Uploader.GetTableSchemaInUpload(tableName)
+	var loadTableStats *types.LoadTableStats
+	var err error
 
-		_, err := pg.loadTable(ctx, tx, tableName, tableSchemaInUpload)
+	err = pg.DB.WithTx(ctx, func(tx *sqlmiddleware.Tx) error {
+		loadTableStats, _, err = pg.loadTable(
+			ctx,
+			tx,
+			tableName,
+			pg.Uploader.GetTableSchemaInUpload(tableName),
+		)
 		return err
 	})
 	if err != nil {
-		return fmt.Errorf("loading table: %w", err)
+		return nil, fmt.Errorf("loading table: %w", err)
 	}
 
-	pg.logger.Infow("completed loading",
+	log.Infow("completed loading")
+
+	return loadTableStats, err
+}
+
+func (pg *Postgres) loadTable(
+	ctx context.Context,
+	txn *sqlmiddleware.Tx,
+	tableName string,
+	tableSchemaInUpload model.TableSchema,
+) (*types.LoadTableStats, string, error) {
+	log := pg.logger.With(
 		logfield.SourceID, pg.Warehouse.Source.ID,
 		logfield.SourceType, pg.Warehouse.Source.SourceDefinition.Name,
 		logfield.DestinationID, pg.Warehouse.Destination.ID,
 		logfield.DestinationType, pg.Warehouse.Destination.DestinationDefinition.Name,
 		logfield.WorkspaceID, pg.Warehouse.WorkspaceID,
 		logfield.Namespace, pg.Namespace,
 		logfield.TableName, tableName,
+		logfield.LoadTableStrategy, pg.loadTableStrategy(),
 	)
+	log.Infow("started loading")
 
-	return nil
-}
+	log.Infow("setting search path")
+	searchPathStmt := fmt.Sprintf(`
+		SET search_path TO %q;
+`,
```

Comment:
[optional] Can't you format it without inserting more bytes into the query that we'll send over with the SQL client?

```suggestion
	searchPathStmt := fmt.Sprintf(`SET search_path TO %q;`,
```